File: be/src/main/java/com/capstone/be/domain/entity/Organization.java
package com.capstone.be.domain.entity;

import com.capstone.be.domain.entity.common.BaseEntity;
import com.capstone.be.domain.enums.OrganizationStatus;
import com.capstone.be.domain.enums.OrganizationType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.util.UUID;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Table(name = "organizations")
@Data
@EqualsAndHashCode(callSuper = true)
public class Organization extends BaseEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;

  @Column(nullable = false, unique = true)
  private String name;

  @Column(nullable = false)
  @Enumerated(EnumType.STRING)
  private OrganizationType type;

  @Column(unique = true, nullable = false)
  private String Name;

  @Column(unique = true, nullable = false)
  private String email;

  @Column(nullable = false)
  private String hotline;

  private String logo;

  @Column(nullable = false)
  private String address;

  @Column(nullable = false)
  private String registrationNumber;

  @Column(nullable = false)
  @Enumerated(EnumType.STRING)
  private OrganizationStatus status = OrganizationStatus.PENDING_VERIFICATION;

  /* ORGANIZATION ADMIN */
  private String adminName;
  private String adminPassword;
  private String adminEmail;

  @Column(nullable = false)
  private Boolean active = true;

  @Column(nullable = false)
  private Boolean deleted = false;

}

File: fe/src/types/organization.ts
import { BaseEntity } from './base';

export enum OrganizationStatus {
  PENDING_VERIFICATION = 'PENDING_VERIFICATION',
  ACTIVE = 'ACTIVE',
  DEACTIVE = 'DEACTIVE',
  DELETED = 'DELETED'
}

export enum OrganizationType {
  TYPE1 = 'TYPE1',
  TYPE2 = 'TYPE2',
  TYPE3 = 'TYPE3'
}

export interface Organization extends BaseEntity {
  id: string;
  userId?: string; // Admin user ID
  organizationId?: string; // Organization profile ID (for statistics API)
  name: string;
  type: OrganizationType;
  email: string;
  hotline: string;
  logo?: string;
  address: string;
  registrationNumber: string;
  status: OrganizationStatus | string;
  adminName?: string;
  adminEmail: string;
  active: boolean;
  deleted: boolean;
  createdAt: string;
  organizationName?: string;
}

export interface OrganizationQueryParams {
  page?: number;
  limit?: number;
  search?: string;
  status?: string;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
  dateFrom?: string;
  dateTo?: string;
}

export interface OrganizationResponse {
  organizations: Organization[];
  total: number;
  page: number;
  limit: number;
}

